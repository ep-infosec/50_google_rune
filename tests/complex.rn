//  Copyright 2021 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

class Complex(self, <x>, <y>) {
  self.x = x
  self.y = y

  func dump(self) {
    println "(", self.x, ", ", self.y, ")"
  }
}

func addComplex(a, b) {
  return Complex(a.x + b.x, a.y + b.y)
}

func mulComplex(a, b) {
  return Complex(a.x * b.x - a.y * b.y, a.x * b.y + a.y * b.x)
}

a = Complex(123i32, 456i32)
b = Complex(1i32, 2i32)
c = Complex(3i32, 4i32)
r1 = mulComplex(a, addComplex(b, c))
r1.dump()

d = Complex(123i64, 456i64)
e = Complex(1i64, 2i64)
f = Complex(3i64, 4i64)
r2 = mulComplex(d, addComplex(e, f))
r2.dump()
