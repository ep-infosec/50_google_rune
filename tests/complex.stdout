(-2244, 2562)
(-2244, 2562)
